hooks/useFetch.js

```diff
@@ -108,31 +112,45 @@ const useFetch = (url, refresh = false) => {
   const [loading, setLoading] = useState(true)
 
   useEffect(() => {
-    const { loading, data, refresh: startRefresh } = fetchData(url)
-    const sub = loading.subscribe(setLoading).add(
-      data.subscribe((ev) => {
-        setData(ev.value.body)
-        setError(ev.error)
-      }),
-    )
+    const fetcher = fetchData(url)
 
-    if (refresh) {
-      sub.add(startRefresh.subscribe())
+    let refresher = fetcher
+    if (!refresh) {
+      refresher = fetcher.pipe(take(1))
     }
 
+    const sub = refresher.subscribe((ev) => {
+      if (ev.ok === false) {
+        setError(ev)
+        setData([])
+        return
+      }
+      setData(ev.body)
+      setError([])
+    })
+
+    of(setLoading(true))
```

Comment:
This is definitely setting `loading: false` a bunch, but it doesn't appear to ever set it to true. I tried adding a couple types events and a `concat`, but that broke everything and I've not figured out why yet.

```js
const refresher = timer(0, 1000).pipe(
    switchMap(() =>
      concat(
        of({ kind: 'loading', state: true }),
        getURL(url).pipe(
          catchError((err) => of(err)),
          publishBehavior(null),
          refCount(),
          filter((ev) => ev !== null),
          map((resp) => ({ kind: 'response', ...resp })),
        ),
        of({ kind: 'loading', state: false }),
      ),
    ),
  )
```